src/music.cpp

```diff
@@ -0,0 +1,86 @@
+#include "music.h"
+#include <fstream>
+#include <iostream>
+
+Asound::Asound() {
+  std::ifstream file(MUSIC_CONFIG_PATH + "config.json");
+
+  if (!file.is_open()) {
+    std::cerr << "Failed to open config file for music" << std::endl;
+    exit(1);
+  }
+
+  file >> config;
+
+  for (auto &[k, v] : config.items()){
+    std::cout << "Read sound key: " << k << " val: " << v << std::endl;
+    std::string s = MUSIC_CONFIG_PATH + std::string(v);
+    sounds.emplace(k, s);
+  }
+}
+
+Asound::~Asound() {}
+
+void Asound::play(std::string sound){
+  sounds[sound].Play();
+}
+
+void Asound::pause(std::string sound) {
```

Comment:
it's really confusing to pause a sound by name.
maybe make play return a sound reference that can be used to pause if necessary (I don't think we'll need to do that often)

Reply:
most of the methods now return `raylib::Sound&` for convenience :+1: